src/frontier/frontier_cmd_arcade.c

```diff
@@ -207,224 +205,224 @@ BOOL FrtCmd_ArcadeAction(FrontierContext *ctx) {
-                    playerMatrix = *(Sprite_GetMatrixPtr(playerSprite));
-
-                    opponentSprite = arcadeCtx->unk60[i]->sprite;
-                    opponentMatrix = *(Sprite_GetMatrixPtr(opponentSprite));
-
-                    Sprite_SetMatrix(playerSprite, &opponentMatrix);
-                    Sprite_SetMatrix(opponentSprite, &playerMatrix);
-                }
-            }
-            break;
-        case 6:
-            sub_02096910(arcadeCtx);
-            break;
-        case 28:
-            *out = BattleArcade_MultiplayerCheck(arcadeCtx->type);
-            break;
-        case 17:
-            *out = arcadeCtx->type;
-            break;
-        case 29:
-            ov80_0222F210(frontierMap);
-            break;
-        case 30:
-            ov80_0222F278(frontierMap);
-            break;
-        case 31:
-            ov80_02234A74(arcadeCtx, frontierMap, var0);
-
+    case 2:
+        arcadeCtx->type = var0;
+        break;
+    case 3:
+        *out = arcadeCtx->unk418[var0];
+        break;
+    case 4:
+        *out = arcadeCtx->winStreak;
+        break;
+    case 5:
+        if (arcadeCtx->winStreak < 9999) {
+            arcadeCtx->winStreak++;
+        }
+        break;
+    case 7:
+        OS_ResetSystem(0);
+        break;
+    case 9:
+        *out = sub_02030E98(arcadeCtx->unk8);
+        break;
+    case 10:
+        ov80_02234588(arcadeCtx, 2);
+        break;
+    case 12:
+        *out = ov80_02238430(arcadeCtx, arcadeCtx->unk13);
+        break;
+    case 14:
+        *out = ov80_02234764(arcadeCtx);
+        break;
+    case 15:
+        if (var0 == 0) {
+            mon = Party_GetMonByIndex(arcadeCtx->playerParty, 0);
+        } else {
+            mon = Party_GetMonByIndex(arcadeCtx->opponentParty, 0);
+        }
+        *out = GetMonData(mon, MON_DATA_HELD_ITEM, NULL);
+        break;
+    case 16:
+        *out = arcadeCtx->unk13;
+        break;
+    case 18:
+        party = SaveArray_Party_Get(param->saveData);
+        for (i = 0; i < 3; i++) {
+            mon = Party_GetMonByIndex(party, arcadeCtx->unk2C[i]);
+            SetMonData(mon, MON_DATA_HELD_ITEM, &arcadeCtx->unk412[i]);
+        }
+        break;
+    case 19:
+        *out = ov80_02238498(arcadeCtx);
+        break;
+    case 20:
+        *out = ov80_02234774(arcadeCtx, var0);
+        break;
+    case 21:
+        ov80_022347B8(arcadeCtx);
+        break;
+    case 22:
+        ov80_022347C4(arcadeCtx);
+        break;
+    case 23:
+        *out = ov80_02234770(arcadeCtx);
+        break;
+    case 24:
+        *out = arcadeCtx->unkA74;
+        break;
+    case 25:
+        if (arcadeCtx->unk13 == 0x1b) {
+            VecFx32 playerMatrix;
+            VecFx32 opponentMatrix;
             u8 playerMonCnt = BattleArcade_GetMonCount(arcadeCtx->type, 1);
-            u8 opponentMonCnt = BattleArcade_GetOpponentMonCount(arcadeCtx->type, 1);
+            BattleArcade_GetOpponentMonCount(arcadeCtx->type, 1);
```

Comment:
Revert this line, it was clearly intended for the return value of the `BattleArcade_GetOpponentMonCount` to be used.